doc/conventions.md

```diff
@@ -0,0 +1,208 @@
+# Conventions
+
+This section contains a set of conventions for handling common situations.
+Each subsection can be read independently.
+
+## Registrations
+
+Components generally need to talk to one another!
+In simple cases, that occurs by method calls.
+But in many cases, a single component needs to communicate with a number of other components that all share some characteristics.
+For example, the `comp/core/health` component monitors the health of many other components, and `comp/workloadmeta /scheduler` provides workload events to an arbitrary number of subscribers.
+
+The convention in the Agent codebase is to use [value groups](./fx.md#value-groups) to accomplish this.
+The _collecting_ component requires a slice of some _registration type_, and the _providing_ components provide that registration type.
+Consider a simple case of a server component to which endpoints can be attached.
+The server is the collecting component, requiring a slice of type `[]*Endpoint`, where `*Endpoint` is the registration type.
+Providing components provide values of type `*Endpoint`.
+
+The collecting component should define the registration type and a constructor for it.
+
+```go
```

Comment:
This example around registration wasn't quite clear to me as `foo/foo.go` seemed disconnected from the `server/` code but that could be just me.

Reply:
It's a different component, but the connection is `server.NewRegistration`.